### PR TITLE
add option to use targetPortName in create service port function

### LINKED ISSUE #11393: [Enhancement]: Service definitions use named ports rather than port numbers
### Related problem

Current Kubernetes best practices suggest use of named ports for connecting services to containers rather than port numbers.  Currently the created services use explicit port numbers when connecting to the generated containers.

### Suggested solution

Use the assigned port names in the created service definitions.

### Alternatives

It currently works.

### Additional context

from the container:
```yaml
      name: kafka
      ports:
      - containerPort: 9090
        name: tcp-ctrlplane
        protocol: TCP
      - containerPort: 9091
        name: tcp-replication
        protocol: TCP
      - containerPort: 9092
        name: tcp-clients
        protocol: TCP
      - containerPort: 9093
        name: tcp-clientstls
        protocol: TCP
      - containerPort: 9094
        name: tcp-external
        protocol: TCP
      - containerPort: 9404
        name: tcp-prometheus
```

from a service:
```yaml
    ports:
    - name: tcp-ctrlplane
      port: 9090
      protocol: TCP
      targetPort: 9090   ### change this
    - name: tcp-replication
      port: 9091
      protocol: TCP
      targetPort: 9091   ### change this
    - name: tcp-kafkaagent
      port: 8443
      protocol: TCP
      targetPort: 8443   ### change this
    - name: tcp-clients
      port: 9092
      protocol: TCP
      targetPort: 9092   ### change this
    - name: tcp-clientstls
      port: 9093
      protocol: TCP
      targetPort: 9093   ### change this
```

### PR DESCRIPTION
**Pull request by KyriosGN0**
### Type of change

_Select the type of your PR_

- Enhancement

### Description

this PR add the option and moved services to use named ports instead of plain port numbers
fixes #11393

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [x] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java [hunk 1]
```diff
@@ -234,17 +236,18 @@ public static ServicePort createServicePort(String name, int port, int targetPor
...
      * @return  Created port
      */
-    public static ServicePort createServicePort(String name, int port, int targetPort, Integer nodePort, String protocol)   {
+    public static ServicePort createServicePort(String name, int port, int targetPort, String targetPortName, Integer nodePort, String protocol)   {
         return new ServicePortBuilder()
                 .withName(name)
                 .withProtocol(protocol)
                 .withPort(port)
-                .withNewTargetPort(targetPort)
+                .withNewTargetPort(Objects.requireNonNullElse(targetPortName, targetPort))
```
Comment (im-konge) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java]: I think we should have another test just for the `targetPort` path - so when the `targetPortName` will be `null`. From the changes I see that only `targetPortName` is checked.

**Review by im-konge — COMMENTED** [review]
Thanks for the PR, I checked the code and have one comment.

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java (hunk 1 above)
Comment (KyriosGN0) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java]: added a test for this scenario

**Review by im-konge — APPROVED** [review]

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java (hunk 1 above)
Comment (scholzj) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java]: Why do we need to have this method handle both `targetPort` and `targetPortName`? I think we should use either one way to configure the services or the other.

Comment (KyriosGN0) [reply cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ServiceUtils.java]: removed the usage of the `targetPort`

**Comment by im-konge** [comment]
@katheris do you have something to this? Or we can merge it?
